### ISSUE TITLE
`@alias` tag does not always work as expected

### ISSUE THREAD

**Issue by hegemonic** [jsdoc3/jsdoc#1385]
When JSDoc parses the following code, the method `Toaster#toast` gets the incorrect longname `<anonymous>~Toaster#toast`:

```js
/**
 * Toaster singleton.
 *
 * @class
 */
var Toaster = (function() {
    var instance = null;

    function Toaster() {}

    /**
     * Toast an item.
     *
     * @alias Toaster#toast
     * @param {BreadyThing} item - The item to toast.
     */
    Toaster.prototype.toast = function(item) {};

    return {
        /**
         * Get the Toaster instance.
         *
         * @alias Toaster.getInstance
         * @returns {Toaster} The Toaster instance.
         */
        getInstance: function() {
            if (instance === null) {
                instance = new Toaster();
                delete instance.constructor;
            }

            return instance;
        }
    };
})();
```

We should probably assume that if you can extract a `memberof` value from the value of the `@alias` tag, then the symbol is a `memberof` that thing, not of `<anonymous>`.

**Comment by hegemonic** [comment]
The following example shows a workaround for this issue:

```js
/**
 * Toaster singleton.
 *
 * @class
 * @hideconstructor
 */
var Toaster = (function() {
    var instance = null;

    function Toaster() {}

    /**
     * Toast an item.
     *
     * @alias toast
     * @memberof Toaster
     * @instance
     * @param {BreadyThing} item - The item to toast.
     */
    Toaster.prototype.toast = function(item) {};

    return {
        /**
         * Get the Toaster instance.
         *
         * @alias Toaster.getInstance
         * @returns {Toaster} The Toaster instance.
         */
        getInstance: function() {
            if (instance === null) {
                instance = new Toaster();
                delete instance.constructor;
            }

            return instance;
        }
    };
})();
```

**Comment by hegemonic** [comment]
Fixed on master. The fix will be included in JSDoc 3.5.2.
